### PR TITLE
Add a note on replication factor value

### PR DESCRIPTION
**Pull request by pierDipi**
We've got a bug [1] report for a misconfiguration of 
`default.topic.replication.factor`.
This PR adds a note that could alert users on the
value of this configuration.

[1] https://github.com/knative-sandbox/eventing-kafka-broker/issues/720

## Proposed Changes 

- Add a note on replication factor value

### CODE REVIEW

File: docs/eventing/broker/kafka-broker.md [hunk 1]
```diff
@@ -105,6 +105,9 @@ The above `ConfigMap` is installed in the cluster. You can edit
 the configuration or create a new one with the same values
 depending on your needs.
 
+_Note: `default.topic.replication.factor` must be less than or equal to the number of
+brokers of your Kafka cluster._
```
Comment (abrennan89) [inline docs/eventing/broker/kafka-broker.md:109]:
```suggestion
**NOTE:** The `default.topic.replication.factor` value must be less than or equal to the number of Kafka broker instances in your cluster. For example, if you only have one Kafka broker, the replication factor should not be more than `1`.
```

Comment (abrennan89) [reply docs/eventing/broker/kafka-broker.md:109]: Suggestion just for formatting and to make this a little easier to understand, up to you if you want to use it :)

**Review by abrennan89 — COMMENTED** [review]
Added a suggestion but it's OK if you want to ignore it @pierDipi :smile: lmk when you need this approved, you can @ me on Slack

File: docs/eventing/broker/kafka-broker.md (hunk 1 above)
Comment (pierDipi) [reply docs/eventing/broker/kafka-broker.md]: Great suggestion! Thanks for making my docs understandable :smile:

Comment (abrennan89) [reply docs/eventing/broker/kafka-broker.md]: No problem! They were already understandable but my tech writer instincts die hard :joy:

**Review by abrennan89 — APPROVED** [review]
/lgtm
Thank you! :smile:

**Comment by abrennan89** [comment]
I consent.

**Comment by abrennan89** [comment]
@googlebot I consent.

**Comment by pierDipi** [comment]
@googlebot I fixed it.

**Comment by grantr** [comment]
All contributors have previously signed the CLA.

**Comment by pierDipi** [comment]
Thanks, Grant :pray: 

/cherry-pick release-0.21

**Pull request merged**